### PR TITLE
python310Packages.amarna: init at 0.1.2

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

This is a static analyzer and linter for https://www.cairo-lang.org/.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/amarna/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    repo = "amarna";
+    rev = "v${version}";
+    sha256 = "sha256-ohR6VJFIvUCMkppqdCV/kJwEmh1fP0QhfQfNu3RoMeU=";
+  };
+
+  propagatedBuildInputs = [
+    lark
+    pydot
+  ];
+
```
Comment (risicle) [inline pkgs/development/python-modules/amarna/default.nix:24]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 186305` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.amarna</li>
    <li>python39Packages.amarna</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 186305` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.amarna</li>
    <li>python39Packages.amarna</li>
  </ul>
</details>

File: pkgs/development/python-modules/amarna/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  propagatedBuildInputs = [
+    lark
+    pydot
+  ];
+  checkInputs = [ pytestCheckHook ];
+  pythonImportsCheck = [ "amarna" ];
+
+  meta = with lib; {
+    description = "Amarna is a static-analyzer and linter for the Cairo programming language.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/amarna/default.nix:28]: Sounds to me like a cli tool which should live outside of pythonPackages.

Comment (RaitoBezarius) [reply pkgs/development/python-modules/amarna/default.nix:28]: You can import `amarna` to reuse it elsewhere and build on the top of it. It is not only a linter, but provides API for static analysis.

**Pull request merged**